### PR TITLE
Add TRMM for diagnostic EDMF

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

Adds TRMM with diagnostic EDMF. Also makes struct MoninObukhov always contain roughness.

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/surface_conditions/surface_setups.jl [hunk 1]
```diff
@@ -101,9 +101,10 @@ function (::Bomex)(params)
     q_vap = FT(0.02245)
     θ_flux = FT(8e-3)
     q_flux = FT(5.2e-5)
+    z0 = FT(1e-4)
```
Comment (trontrytel) [inline src/surface_conditions/surface_setups.jl:104]: why not zero?

Comment (szy21) [reply src/surface_conditions/surface_setups.jl:104]: You can't use zero as it is used in the denominator in surface flux. 1e-4 for roughness is a typical value over ocean.

Comment (trontrytel) [reply src/surface_conditions/surface_setups.jl:104]: Or maybe it's a different variable than what I think it is/ Maybe a longer name?

**Review by trontrytel — APPROVED** [review]

File: src/surface_conditions/surface_setups.jl (hunk 1 above)
Comment (szy21) [reply src/surface_conditions/surface_setups.jl:104]: Right, it is roughness, and it is commonly noted as z0 in Monin-Obukhov.

Comment (szy21) [reply src/surface_conditions/surface_setups.jl:104]: I can add a comment on the MoninObukhov struct to make this more clear, if that's helpful?

**Pull request merged**